partner_communication/models/communication_job.py

```diff
@@ -522,10 +633,32 @@ def _print_report(self):
         for job in self:
             # Get pdf should directly send it to the printer if report
             # is correctly configured.
-            report_obj.with_context(
+            pdf_data = report_obj.with_context(
                 print_name=self.env.user.firstname[:3] + ' ' + (
-                    job.subject or '')
+                    job.subject or ''),
+                must_skip_send_to_printer=True
             ).get_pdf(job.ids, job.report_id.report_name)
+
+            # add omr to pdf if needed
+            if job.omr_enable_marks:
+                is_latest_document = False if \
+                    len(self.attachment_ids) > 0 \
+                    and len(self.attachment_ids.filtered(
+                        lambda r: r.attachment_id.disable_omr == False)) > 0 \
```

Comment:
Here you could simplify `is_latest_document = not self.attachment_ids.filtered('enable_orm')`
and I propose that you change the field `disable_orm` to `enable_orm` so that you can use the filtered function with just the name of the field.

Reply:
I change the name to `enable_omr` and the filter should be `.filtered('attachment_id.enable_omr')`